python-threatexchange/threatexchange/exchanges/impl/ncmec_api.py

```diff
@@ -303,16 +330,23 @@ def log(event: str) -> None:
                         # Our entry estimatation (based on the cursor parameters)
                         # occasionally seem to over-estimate
                         log(f"est {entry.estimated_entries_in_range} entries")
-                elif i % 100 == 0:
-                    # If we get down to one second, we can potentially be
-                    # fetching an arbitrary large amount of data in one go,
-                    # so log something occasionally
-                    log(f"large fetch ({i}), up to {len(updates)}")
-                updates.extend(entry.updates)
+
+                if i % 100 == 5:
+                    # On large fetches, log notice every once in a while
+                    log(f"large fetch ({i}) with {len(entry.updates)} updates.")
+
+                yield state.FetchDelta(
+                    {f"{entry.member_id}-{entry.id}": entry for entry in entry.updates},
+                    NCMECCheckpoint(
+                        get_entries_max_ts=current_start,
+                        _paging_url=entry.next,
+                    ),
+                )
+
             else:  # AKA a successful fetch
                 # If we're hovering near the single-fetch limit for a period
                 # of time, we can likely safely expand our range.
-                if len(updates) < api.NCMECHashAPI.ENTRIES_PER_FETCH * 2:
+                if len(entry.updates) < api.NCMECHashAPI.ENTRIES_PER_FETCH * 2:
```

Comment:
blocking: entry will be unset if you if the for loop doesn't yield. I'm surprised typing doesn't catch this. 

Here's a demonstrator:
```
$ python
>>> items = []
>>> for i, val in items:
...   print(i, val)
... else:
...   print(val)
... 
Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
NameError: name 'val' is not defined.
```

To fix, on L300 add
```
entry = None # Alternatively, set it to a empty entry object
```

and then all these checks become 
```
if (entry is not None and len(entry.updates) ...
```